app/components/edit_resource_button.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# EditResourceButton
+# edit collection btn + update work btn
+# pass in resource instead of work_version and infer type (collection or work_version)
+
+class EditResourceButton < ApplicationComponent
+  attr_reader :resource, :policy
+
+  def initialize(resource:, policy:)
+    @resource = resource
+    @policy = policy
+  end
+
+  private
+
+    def path
+      if collection?
+        dashboard_form_collection_details_path(resource.id)
+      elsif has_draft?
+        # update the existing draft version
+        dashboard_form_work_version_details_path(resource.work.draft_version.id)
+      elsif policy.new?
+        # create a new draft work version
+        dashboard_work_work_versions_path(resource.work)
+      end
+    end
+
+    def label
+      collection? ? I18n.t('resources.collection.edit_button') : I18n.t('resources.work_version.edit_button')
```

Comment:
There's a feature in ActiveRecord models where you can get the i18n key. It's also how it builds all the `edt_*_path` methods and other things. See https://api.rubyonrails.org/classes/ActiveModel/Name.html

Try `I18n.t("resources.#{resource.model_name.i18n_key}.edit_button")`

Even though the resource is a decorated version of the active record object, any methods not explicitly defined on the decorated model, just get pushed up to the model that it's decorating. So we have access to all the same methods we do if resource was just a Collection or WorkVersion.